src/diffpy/utils/tools.py

```diff
@@ -30,31 +68,31 @@ def _sorted_merge(*dicts):
 
 def _create_global_config(args):
     username = input(f"Please enter the name of the user to put in the diffpy global config file "
-                     f"[{args.get("username", "")}]:  ").strip() or args.get("username", "")
+                     f"[{args.get('username', '')}]:  ").strip() or args.get("username", "")
     email = input(f"Please enter the email of the user to put in the diffpy global config file "
-                     f"[{args.get("email", "")}]:  ").strip() or args.get("email", "")
+                     f"[{args.get('email', '')}]:  ").strip() or args.get("email", "")
     return_bool = False if username is None or email is None else True
     with open(Path().home() / "diffpyconfig.json", "w") as f:
         f.write(json.dumps({"username": stringify(username), "email": stringify(email)}))
     return return_bool
 
 def get_user_info(args=None):
     """
-    Get username and email.
+    get username and email configuration
 
-    Prompt the user to enter username and email.
-    If not provided, read from the config file (first from cwd, and then from home directory).
-    If neither are available, raise a ValueError.
-    Save provided values to the config file if a config file doesn't exist.
+    First attempts to load config file from global and local paths.
+    If neither exists, creates a global config file.
+    It prioritizes values from args, then local, then global.
+    Removes invalid global config file if creation is needed, replacing it with empty username and email.
 
     Parameters
     ----------
     args argparse.Namespace
-        the arguments from the parser
+        the arguments from the parser, default is None
```

Comment:
I think I made a mistake here. The args here is a dictionary instead of a Namespace. Would we want args to be a dictionary or Namespace here?

Reply:
We could use either, but I thought dict was more general as this is, in principle, something that we want to reuse in different contexts, so I thought it was better not to expect that there was a cli in general.  When there is a cli (as in labpdfproc) we can turn the argparse.Namespace into a dict when we pass it in.

It could have been done the other way, I just made a choice here.